sympy/integrals/risch.py

```diff
@@ -218,14 +218,14 @@ def __init__(self, f=None, x=None, handle_first='log', dummy=False, extension=No
         self.dummy = dummy
         self.reset()
         exp_new_extension, log_new_extension = True, True
-        if rewrite_complex:
+        if rewrite_complex or I in self.f.atoms():
```

Comment:
This could be included in the definition of rewrite complex. (Set to True if `I in self.f.atoms()`.)

Reply:
I would set the default value to `None`, meaning "automatic" (it does it if it detects a complex constant, and not otherwise). It can also be set to True or False manually by the user (with a caveat that False could lead to invalid nonelementary results).